lib/chef/util/dsc/local_configuration_manager.rb

```diff
@@ -72,10 +70,22 @@ def run_configuration_cmdlet(configuration_document, apply_configuration, shello
       status
     end
 
-    def lcm_command_code(configuration_path, test_only_parameters)
-      <<-EOH
-$ProgressPreference = 'SilentlyContinue';start-dscconfiguration -path #{@configuration_path} -wait -erroraction 'stop' -force #{test_only_parameters}
-EOH
+    def lcm_command(apply_configuration)
+      @common_command_prefix = "$ProgressPreference = 'SilentlyContinue';"
+      ps4_base_command = "#{@common_command_prefix} Start-DscConfiguration -path #{@configuration_path} -wait -erroraction 'stop' -force"
+      apply_configuration ? ps4_base_command : test_lcm_configuration(ps4_base_command)
+    end
+
+    def test_lcm_configuration(ps4_base_command)
+      if ps_version_gte_5?
+        "#{@common_command_prefix} Test-DscConfiguration -path #{@configuration_path}"
```

Comment:
We are using an instance variable `@common_command_prefix` here which is required to be set prior to calling this method. If we need to use this `test_lcm_configuration` method for any other command in future then we'll have to make sure to define `@common_command_prefix` otherwise the function will throw error for PS 5.
It would be good to pass `common_command_prefix` as an argument.

Reply:
` test_lcm_configuration` is only called from `lcm_command`. I'd combine it back into `lcm_command` similar to how I had https://github.com/chef/chef/commit/ca92458609a6a475eb69f6b94aac3dcd71e6ecbe. That way you don't have to set `@common_command_prefix` or pass `ps4_base_command` to a method that only needs it half the time.